### DISCUSSION TITLE
sqlite3 version?

### DISCUSSION THREAD

**Discussion by sqllyw** [sqlitebrowser/sqlitebrowser#2989] [Q&A]
Hi
what is the version of sqlite3 built in the browser? thanks

**Comment by chrisjlocke** [comment]
Would you believe me if I told you there are three answers?  Well, actually, there are probably about nine answers.
It all depends on the operating system, and which build you're referring to.  For the release build (3.12.2), it was 3.35.5, but the SQLCipher version of DB4S uses SQLCipher 4.4.3 which was based off SQLite 3.34.1.
The nightlies are more up to date, with the latest nightly using 3.37.2.  The SQLCipher version in the nightly is SQLCipher 4.5, which is based off SQLite 3.36.0.
If you're using Linux, then the SQLite version depends on what's installed on your system, as it uses the system SQLite, and not what is bundled with DB4S.

Clear as mud? 😂😂

**Comment by justinclift** [comment]
For the Windows nightly builds, they were updated to use the newer SQLite 3.38.0 a few days after that was released.

Looking at the SQLite.org website just now, there's a newer 3.38.1 release available, so I'm updating our Windows nightly builds to use that from tonight onwards as well. :smile:

**Reply by sqllyw** [reply]
I'm using Linux, upgraded sqlite3 to 3.38:
````
sqlite3 --version
3.38.1 2022-03-12 13:37:29 38c210fdd258658321c85ec9c01a072fda3ada94540e3239d29b34dc547a8cbc
````
but sqlitebrowser still shows 3.8:
````
select sqlite_version()
3.8.6
````
in the About:
````
DB Browser for SQLite Version 3.12.2

Built for x86_64-little_endian-lp64, running on x86_64
Qt Version 5.15.2
SQLCipher Version 3.2.0 (based on SQLite 3.8.6).
````

**Reply by justinclift** [reply]
> upgraded sqlite3 to 3.38

Check if your Linux distribution has a later version of SQLCipher available too.  That's what our application is using on your system, and it's currently pretty old (SQLCipher 3.2.0, based on SQLite 3.8.6). :smile:

**Comment by chrisjlocke** [comment]
Aah, the SQLCipher version of DB4S doesn't use SQLite, as it uses SQLCipher (for encryption).  Its a different library.
There is a version of DB4S that uses SQLite rather than SQLCipher, or you can build it yourself.  
What distro of Linux are you using?  Worth having a look at this issue, which sort of covers the same ground, and mentions a PPA to get another version.
https://github.com/sqlitebrowser/sqlitebrowser/issues/2610
